qiskit_experiments/library/quantum_volume/qv_analysis.py

```diff
@@ -15,17 +15,123 @@
 
 import math
 import warnings
-from typing import Optional
+from typing import List
 
 import numpy as np
 import uncertainties
 from qiskit_experiments.exceptions import AnalysisError
-from qiskit_experiments.curve_analysis.visualization import plot_scatter, plot_errorbar
 from qiskit_experiments.framework import (
     BaseAnalysis,
     AnalysisResultData,
     Options,
 )
+from qiskit_experiments.visualization import BasePlotter, MplDrawer
+
+
+class QuantumVolumePlotter(BasePlotter):
+    """Plotter for QuantumVolumeAnalysis"""
+
+    @classmethod
+    def expected_series_data_keys(cls) -> List[str]:
+        """Returns the expected series data keys supported by this plotter.
+
+        Data Keys:
+            hops: Heavy-output probability fraction for each circuit
+        """
+        return ["hops"]
+
+    @classmethod
+    def expected_supplementary_data_keys(cls) -> List[str]:
+        """Returns the expected figures data keys supported by this plotter.
+
+        Data Keys:
+            depth: The depth of the quantun volume circuits used in the experiment
+        """
+        return ["depth"]
+
+    def set_supplementary_data(self, **data_kwargs):
+        """Sets supplementary data for the plotter.
+
+        Args:
+            data_kwargs: See :meth:`expected_supplementary_data_keys` for the
+                expected supplementary data keys.
+        """
+        # Hook method to capture the depth for inclusion in the plot title
+        if "depth" in data_kwargs:
+            self.set_figure_options(
+                figure_title=(
+                    f"Quantum Volume experiment for depth {data_kwargs['depth']}"
+                    " - accumulative hop"
+                ),
+            )
+        super().set_supplementary_data(**data_kwargs)
+
+    @classmethod
+    def _default_figure_options(cls) -> Options:
+        options = super()._default_figure_options()
+        options.xlabel = "Number of Trials"
+        options.ylabel = "Heavy Output Probability"
+        options.figure_title = "Quantum Volume experiment - accumulative hop"
+        options.series_params = {
+            "hop": {"color": "gray", "symbol": "."},
+            "threshold": {"color": "black", "linestyle": "dashed", "linewidth": 1},
+            "hop_cumulative": {"color": "r"},
+            "hop_twosigma": {"color": "lightgray"},
+        }
+        return options
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        options = super()._default_options()
+        options.style["figsize"] = (6.4, 4.8)
+        options.style["axis_label_size"] = 14
+        options.style["symbol_size"] = 2
+        return options
+
+    def _plot_figure(self):
+        series = self.series[0]
```

Comment:
Do we want to limit the quantum volume plots to only one series? I'm okay with that limitation if it makes sense for quantum volume plots. If we don't support multiple series, then we need to document this somewhere, like in the class docstring.

Reply:
The plot is pretty busy, so I think we do want to limit to one series:

![image](https://github.com/Qiskit-Extensions/qiskit-experiments/assets/1178780/ea1696cb-c883-429a-ace7-e8e4acecc4ef)

The original code only plotted one series and the goal here was to move the old plotter to the new system, so we could finish deprecating the old system (it was the only thing left using the old plotter code). I think we could make a version that handled more than one series but that could be left as a follow up. We would probably want to change the presentation since overlapping shaded regions would not work well. Maybe just plot lines for the -2sigma boundary instead of shading the region.

I think I handled this part strangely. I limited the plotter to one unnamed series -- why not just name it? I did that in [79404db](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1348/commits/79404db910a1dccd26366e541b69c10e5eacd672) along with documenting that series as the only one accepted.

Reply:
That makes lots of sense, especially given it's blocking the deprecation of old plotting code. All of your changes look good to me. Thanks for the quick update to the PR!